server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -207,6 +213,22 @@ public enum MetadataVersion {
      */
     public static final MetadataVersion MINIMUM_BOOTSTRAP_VERSION = IBP_3_3_IV0;
 
+    /**
+     * The latest production-ready MetadataVersion. This is the latest version that is stable
+     * and cannot be changed. MetadataVersions later than this can be tested via junit, but
+     * not deployed in production.
+     *
+     * <strong>Think carefully before you update this value. ONCE A METADATA VERSION IS PRODUCTION,
+     * IT CANNOT BE CHANGED.</strong>
+     */
+    public static final MetadataVersion LATEST_PRODUCTION = IBP_3_7_IV0;
```

Comment:
So I guess we need another JIRA to change this to `IBP_3_7_IV2` before the 3.7 release?

Reply:
yes. once it's production we'll change this (could be with a MINOR or with a JIRA). And then we can't change it